File: src/web/LoginServlet.java
package web;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.*;
import java.io.IOException;

@WebServlet(name = "web.LoginServlet", urlPatterns = "/login.do")
public class LoginServlet extends HttpServlet {
  protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
    final String input_username = request.getParameter("username");
    // final String input_password = request.getParameter("password");
    // final String input_auto_login = request.getParameter("auto_login");

    if (input_username != null /*&& input_password != null*/) {
      final HttpSession session = request.getSession();
      // if (input_auto_login != null) {
      //   final Cookie login_username_cookie = new Cookie("username", input_username);
      //   login_username_cookie.setMaxAge(30 * 60);
      //   response.addCookie(login_username_cookie);
      //   session.setAttribute("username", input_username);
      // }
      final Cookie login_username_cookie = new Cookie("username", input_username);
      response.addCookie(login_username_cookie);
      session.setAttribute("username", input_username);
      response.sendRedirect("/list");
    }
  }
}

File: src/web/LoginServlet.kt
package web

import javax.servlet.ServletException
import javax.servlet.annotation.WebServlet
import javax.servlet.http.*
import java.io.IOException

@WebServlet(name = "web.LoginServlet", urlPatterns = arrayOf("/login.do"))
class LoginServlet : HttpServlet() {
  @Throws(ServletException::class, IOException::class)
  override fun doPost(request: HttpServletRequest, response: HttpServletResponse) {
    val inputUsername: String = request.getParameter("username") // 用户名
    val inputRemember: String? = request.getParameter("remember")
    val isRemember: Boolean = !inputRemember.isNullOrBlank() // 是否自动登陆
    println("inputUsername = ${inputUsername}")
    val session = request.session
    val cookieUsername = Cookie("username", inputUsername)
    response.addCookie(cookieUsername)
    session.setAttribute("username", inputUsername)
    if (isRemember)
      response.addCookie(Cookie("remember", "1"))

    response.sendRedirect("/list")
  }
}

